File: 5_deepseek_sol2.py
MOD = 998244353

T = int(input())

for _ in range(T):
    N = int(input())
    numbers = input().split()
    
    from functools import cmp_to_key
    
    def compare(a, b):
        if a + b > b + a:
            return -1
        elif a + b < b + a:
            return 1
        else:
            return 0
    
    numbers.sort(key=cmp_to_key(compare))
    
    final_number = ''.join(numbers)
    
    answer = 0
    for digit_char in final_number:
        digit = int(digit_char)
        answer = (answer * 10 + digit) % MOD
    
    print(answer)

File: 5_deepseek_sol2.java
import java.io.*;
import java.util.*;

public class Main {
    static final int MOD = 998244353;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        
        while (T-- > 0) {
            int N = Integer.parseInt(br.readLine());
            String[] numbers = br.readLine().split(" ");
            
            List<String> list = new ArrayList<>();
            for (String num : numbers) {
                list.add(num);
            }
            
            Collections.sort(list, new Comparator<String>() {
                @Override
                public int compare(String a, String b) {
                    String order1 = a + b;
                    String order2 = b + a;
                    return order2.compareTo(order1);
                }
            });
            
            StringBuilder result = new StringBuilder();
            for (String num : list) {
                result.append(num);
            }
            
            long answer = 0;
            String finalNumber = result.toString();
            for (int i = 0; i < finalNumber.length(); i++) {
                int digit = finalNumber.charAt(i) - '0';
                answer = (answer * 10 + digit) % MOD;
            }
            
            System.out.println(answer);
        }
    }
}
